boa_tester/src/main.rs

```diff
@@ -241,12 +241,45 @@ fn main() -> Result<()> {
     }
 }
 
+fn clone_test262() {
+    const TEST262_REPOSITORY: &str = "https://github.com/tc39/test262";
+    const TEST262_COMMIT: &str = "0c87a86b58391b40aa7623b919603d87d4b77a4d";
+    const TEST262_DIRECTORY: &str = "test262";
+
+    if Path::new(TEST262_DIRECTORY).is_dir() {
+        return;
+    }
+
+    println!("Cloning test262");
+    let result = Command::new("git")
+                    .arg("clone")
+                    .arg(TEST262_REPOSITORY)
+                    .arg(TEST262_DIRECTORY)
+                    .status()
+                    .expect("");
+
+    println!("result: {result}");
+    assert!(result.success());
+
+    println!("reset to known-good rev");
+    let result = Command::new("git")
+                    .arg("reset")
+                    .arg("--hard")
+                    .arg(TEST262_COMMIT)
```

Comment:
We could just default to the latest commit in `main`, and provide a commit from the CLI. We could then hard-code this in our CI if needed.

Reply:
agree, it seems odd to choose an arbitary commit. Why did we choose this one? Was it just the one we're using at the moment?

 I guess CI would have some environment variable to swap "main" out with whatever we're using.

Reply:
Agreed - seems fine to default to "main" and override via CLI which sources the value from CI or other source the user controls.